### PR TITLE
kotlin: 1.4.21 -> 1.4.32

### PR DESCRIPTION
**Pull request by 239**
###### Motivation for this change
Updated Kotlin compiler to 1.4.32

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119410` at f006f346 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>kotlin</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/kotlin/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/kotlin/default.nix:42:5:
```
   |
42 |     maintainers = with lib.maintainers;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/compilers/kotlin/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
 { lib, stdenv, fetchurl, makeWrapper, jre, unzip }:
 
 let
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/kotlin/default.nix:3]:
```suggestion
```

File: pkgs/development/compilers/kotlin/default.nix [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-  version = "1.4.21";
+  version = "1.4.32";
 in stdenv.mkDerivation {
   inherit version;
   pname = "kotlin";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/kotlin/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {
  pname = "kotlin";
  version = "1.4.32";
```

**Comment by 239** [comment]
@SuperSandro2000 Should I close this and submit the update to 1.5.0 instead?

**Pull request merged**